protocol/superchainerc20/L1-native-factory.md

```diff
@@ -0,0 +1,170 @@
+
+Both these problems can be solved by introducing factories that comply with the following requirements:
+
+- **Same address deployment:** the `SuperchainERC20` standard requires the same address across chains.
+  - Notice this is not strictly necessary: allowing different addresses of `SuperchainERC20`s to communicate using a registry is possible. However, maintaining a registry would be a demanding task.
+- **Implementation upgradability:** it should be easy to upgrade `OptimismSuperchainERC20` implementations across chains.
+- **Factory upgradability:** The factory should be upgradable to follow OP Factories structure.
+- **Deployment history:** The factory should keep a history of deployed tokens, acting as a registry.
+
+> 💡
+> The presented solution is focused on tokens that are native to L1. For L2 native tokens, the factory implementation should be modified.
+
+## Proposed Solution
+
+In what follows, we will list each requirement and detail how the factory should look to accomplish them.
+
+### Same address: Predeploys + `CREATE3`
+
+**Predeploys**
+
+Factories will live in each L2 as a predeploy on the same address, which is crucial to achieving the same address for `OptimismSuperchainERC20`s across chains.
+
+If the same address for the `OptimismSuperchainERC20` on L1 is also needed, then the address at which the predeploy is initialized should match the factory's L1 address.
+
+**Creation method**
+
+The `OptimismSuperchainERC20` addresses should be independent of the implementations. `CREATE` was discarded due to nonce dependability, and `CREATE2` was discarded because of its dependence on the creation code. For these reasons, the factory should use `CREATE3`.
+
+The salt will be composed of the L1 address and the token metadata. This implies that the same L1 token can have multiple SuperchainERC20 representation, as long as the metadata also changes.
+
+### Implementation Upgradability: Beacon Proxies
+
+**BeaconProxies**
+
+The factory will deploy `OptimismSuperchainERC20`s as BeaconProxies, as this is the easiest way to upgrade multiple proxies simultaneously. Each BeaconProxy will delegate-call to the implementation address provided by the Beacon Contract.
+
+**Beacon Contract**
+
+The Beacon Contract holds the implementation address for the BeaconProxies. It will be a predeploy (consistent with the current architecture of the OP stack).
+
+**Implementations**
+
+Each type of ERC20 (vanilla, vote token, rebasing, xERC20, etc) will require their unique implementation and, therefore, factory. Optimism should deploy and support the most common implementations. Anyone can deploy their custom implementations and factories but without predeploys.
```

Comment:
We should scope this specifically to the `OptimismSuperchainERC20` factory and then we can revisit designs for more generic factories in the future